network/src/protocol/mod.rs

```diff
@@ -1192,6 +1198,33 @@ async fn statement_import_loop<Api>(
 	}
 }
 
+fn fetch_pov_from_gossip(
+	candidate: &AbridgedCandidateReceipt,
+	gossip_handle: &impl GossipOps,
+) -> impl Future<Output = PoVBlock> + Send {
+	let candidate_hash = candidate.hash();
+	let topic = crate::legacy::gossip::pov_block_topic(candidate_hash);
+
+	// The gossip system checks that the correct pov-block data is present
```

Comment:
If that is checked, why do we need to check the candidate hash again?

Reply:
the topic should be instantiated by relay-parent, not candidate hash. that's a bug I have now added a test for